src/views/course_pairings.vue

```diff
@@ -0,0 +1,212 @@
+         * to populate the page with the specified course.
+         */
+        specify_course: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when specify_term is modified. The data is used
+         * to populate the page with the specified term.
+         */
+        specify_term: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when column is modified. The data is used
+         * to populate the page with the specified column.
+         */
+        column: function() {
+            this.getData();
+        }
+    },
+    methods: {
+        /**
+         * Function gets course pairing data from API for the specified
+         * course, term and column and calls parseData to parse retuned data.
+         */
+        getData: function() {
+            // Converting to parameter enums
+            var column_name = this.column === "TA" ? "ta" : "instructor";
+
+            var course_where = "";
+            var term_where = "";
+            if (this.specify_term != "All Terms") {
+                term_where = `${this.specify_term}`;
+            }
+            if (this.specify_course != "All Courses") {
+                course_where = `${this.specify_course}`;
+            }
+
+            let url = `{
+                "what":"course_pairings",
+                "user_id":"${this.$route.params.user_id}",
+                "term":"${term_where}",
+                "course_code":"${course_where}",
+                "column":"${column_name}"
+                }`;
+
+            url = JSON.parse(url);
+            fetch("get_info.php?" + generate_query_string(url))
+                .then(res => res.json())
+                .then(data => this.parseData(data))
+                .catch(err => {
+                    this.$emit("error", err.toString());
+                });
+        },
+
+        /**
+         * Function parses data from API call that will be rendered on the
+         * spreadsheet and select component.
+         */
+        parseData: function(data) {
+            var pairings = {};
+            for (let element of data.DATA) {
+                // Create an associative array with keys as course code and values
+                // as an array of instructors/tas for that couse.
+                pairings[element.course_code] =
+                    pairings[element.course_code] || [];
+                pairings[element.course_code].push({
+                    name: element.name,
+                    user_id: element.user_id
+                });
+
+                // Pushing list of distinct terms
+                if (this.list_terms.indexOf(element.term) < 0) {
+                    this.list_terms.push(element.term);
+                }
+            }
+            var user_associations = Object.keys(pairings).map((course)=>{
+                if (this.courses_list.indexOf(course) < 0) {
```

Comment:
Is it important that this is a list? If you're doing this check a lot, it makes more sense for it to be an object and to query for keys.

```
for (let term of terms) {
     this.terms[term] = true
}
if (this.terms[queryTerm]) { ... }
```